### PR TITLE
feat: add package.xml

### PR DESCRIPTION
**Pull request by wep21**
- add package.xml for ros

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  <maintainer email="k.koide@aist.go.jp">Kenji Koide</maintainer>
+  <license>MIT</license>
+
+  <buildtool_depend>cmake</buildtool_depend>
+  <depend>eigen</depend>
+  <depend>libomp-dev</depend>
+  <depend>libpcl-all-dev</depend>
+  <depend>nanoflann</depend>
+  <depend>sophus</depend>
```
Comment (valgur) [inline package.xml:16]:
```suggestion
```
Both nanoflann and Sophus are vendored with the library and significantly extended/customized.

**Comment by koide3** [comment]
Might be better to drop the dependency on PCL and TBB as well to avoid forcing people to install the huge dependency.

I'm wondering if we should directly add package.xml to small_gicp or we should create a separate vendor package. I tend to want to keep a package isolated from ROS when it's possible.

 @valgur Do you have any opinions?

**Comment by koide3** [comment]
I think it would be reasonable to add `package.xml` because many ROS people would be interested in this package. I would like to merge this PR once  `libpcl-all-dev` and `tbb` are removed.

**Comment by valgur** [comment]
The library is definitely relevant to ROS, so a ROS 1/2 packaging support directly in the repo would be welcome, I think.

However, the current PR is incomplete for a catkin package. It's missing `find_package(catkin ...)`, `catkin_package()` and catkin-specific `install()` commands. Creating a separate `cmake/catkin.cmake` that could then be conditionally included might make sense to keep things clean. The install commands in the main CMakeLists.txt can be made configurable, so they can be overridden for catkin, perhaps.

I would also strongly encourage adding a minimal CI workflow to test this.

Might be worth looking into adding ROS 2 / Ament support at the same thime as well.

**Comment by wep21** [comment]
make pcl and tbb optional in package.xml at https://github.com/koide3/small_gicp/pull/26/commits/9ff9b1b68d44d15536f9e80272a8cecaeff55a9e.

**Comment by wep21** [comment]
I guess [This minimum change](https://github.com/OctoMap/octomap/blob/eff7d05c23f95d1b1492d6e79523e74e64adc87f/octomap/CMakeLists.txt#L72-L77) make the package behave as both catkin/ament package.

**Comment by koide3** [comment]
Thanks. I think we can merge the PR in the current form for now and add further modifications later.

**Pull request merged**